components/library/libtorrent-rasterbar/Makefile

```diff
@@ -0,0 +1,67 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2022, Daniel Chan
+#
+
+BUILD_BITS=			64
+BUILD_STYLE=	cmake
+include ../../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		libtorrent-rasterbar
+COMPONENT_VERSION=	2.0.5
+COMPONENT_PREP_GIT=yes
+GIT_TAG=v$(COMPONENT_VERSION)
```

Comment:
Please use fixed versions if possible. Cloning git repos is just acceptable as the last resort.
You can find download URIs for it here:
https://github.com/arvidn/libtorrent/tags
(megasync is a major pain - it gets republished with every jenkins run and should be changed to a fixed version, too.)

Reply:
It has several submodules need to be cloned. Anyway to do multiple download in the same Makefile?

Reply:
Most problems aren't new and have already been solved in the past. My general recommendation is to search for them and look how they have been solved. One example: cd to the components folder and then start a search similar to this:
find . -name Makefile -exec grep ARCHIVE_1 {} \; -print
This will give you some Makefiles where additional sources have been incorporated. Study a few of them.
Another source of information is the contents of make-rules folder.

Reply:
Fixed,